hubclient/client.go

```diff
@@ -206,25 +221,40 @@ func (c *Client) HttpGetJSON(url string, result interface{}, expectedStatusCode
 	}
 
 	err = c.processResponse(resp, result, expectedStatusCode)
-	return AnnotateHubClientErrorf(err, "unable to process response from GET to %s", url)
+	return AnnotateHubClientErrorf(err, "unable to process response from GET to %s", url), resp
+}
+
+func (c *Client) HttpPutString(url string, data string, contentType string, expectedStatusCode int) error {
+	return c.httpPut(url, data, contentType, expectedStatusCode, false)
 }
 
 func (c *Client) HttpPutJSON(url string, data interface{}, contentType string, expectedStatusCode int) error {
+	return c.httpPut(url, data, contentType, expectedStatusCode, true)
+}
+
+func (c *Client) httpPut(url string, data interface{}, contentType string, expectedStatusCode int, isJsonData bool) error {
 	var resp *http.Response
 
 	if c.debugFlags&HubClientDebugTimings != 0 {
 		log.Debugf("DEBUG HTTP STARTING %s REQUEST: %s", http.MethodPut, url)
 	}
 
-	// Encode json
-	var buf bytes.Buffer
-	enc := json.NewEncoder(&buf)
+	var reader io.Reader
 
-	if err := enc.Encode(&data); err != nil {
-		return newHubClientError(nil, nil, fmt.Sprintf("error encoding json: %+v", err), err)
+	if isJsonData {
```

Comment:
feels like it would be more appropriate to create 2 functions instead of passing a flag, and extract common part below into separate function.